CDCgov/prime-reportstream
# Oo/oslynn/10995/add support for nbs to rest transport

This PR add support for CDC NBS to REST Transport.

Test Steps:
1. clone the repos
2. ./gradlew clean package
3. download the attached file and rename it to .hl7
[pdi-covid-19-ks.txt](https://github.com/CDCgov/prime-reportstream/files/12413258/pdi-covid-19-ks.txt)
4. update the setting ./prime multiple-settings set --input ./settings/staging/0154-nbs-staging.yml
5. docker-compose up
6. curl -X POST -H "client:primary" -H "content-type:application/hl7-v2" --data-binary "@./pdi-covid-19-ks.hl7" "http://localhost:7071/api/reports"
7. Note report Id (my is: d551f3c4-1b01-4812-a209-b3ef9fd7c114)
8. Check the history
9.  curl "http://localhost:7071/api/waters/report/d551f3c4-1b01-4812-a209-b3ef9fd7c114/history"
![image](https://github.com/CDCgov/prime-reportstream/assets/20068283/dfdc82a4-3d4a-43d6-84f9-3bf7172942f2)

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue
- https://app.zenhub.com/workspaces/engagement-6166edbd409257001e09f1b2/issues/gh/cdcgov/prime-reportstream/10995

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
